src/BSON/Value.c

```diff
@@ -53,15 +54,23 @@ static HashTable* php_phongo_value_get_properties_hash(phongo_compat_object_hand
 
 	intern = Z_OBJ_VALUE(PHONGO_COMPAT_GET_OBJ(object));
 
-	PHONGO_GET_PROPERTY_HASH_INIT_PROPS(is_temp, intern, props, 1);
+	PHONGO_GET_PROPERTY_HASH_INIT_PROPS(is_temp, intern, props, 2);
 
-	// TODO: Print value (somehow)
-	//{
-	//	zval data;
-	//
-	//	ZVAL_STR(&data, php_base64_encode((const unsigned char*) bson_get_data(intern->bson), intern->bson->len));
-	//	zend_hash_str_update(props, "data", sizeof("data") - 1, &data);
-	//}
+	{
+		zval type;
+
+		ZVAL_STRING(&type, php_phongo_bson_type_to_string(intern->value.value_type));
+
+		zend_hash_str_update(props, "type", sizeof("type") - 1, &type);
+	}
+
+	{
+		zval value;
+
+		php_phongo_bson_value_to_zval(&intern->value, &value);
+
+		zend_hash_str_update(props, "value", sizeof("value") - 1, &value);
```

Comment:
Note that the `get_properties_hash` function typically corresponds to `init_from_hash`, and both are used for `__set_state` and PHP serialization in addition to debug output. That could be problematic here:

 * There will need to be an inverse of `php_phongo_bson_type_to_string`
 * Relying on `php_phongo_bson_value_to_zval` means we might have trouble round-tripping BSON int64 (regardless of system architecture).

Reply:
Serialisation is something I've left out as per the original comment in the stub class. While I think we can avoid thinking about BSON serialisation, regular PHP serialisation is something we should implement. The only issue here is int64 with 32-bit values, which might be solved if `phongo_bson_value_to_zval` always returns an `Int64` instance for the type.

That said, I'll work on `init_from_hash` next and make sure that serialisation of types works.

Reply:
If we're dealing with PHP serialization and `var_export()`, I think it'd be sensible to output a base64 binary string (i.e. raw BSON in an envelope). I don't see any reason to rely on zval conversion, and it'd hopefully make the `init_from_hash` function much easier to implement. There, we probably wouldn't even need to output the type code at all.

This would mean the `get_debug_info` handler would need separate logic, assuming you'd like `var_dump()` to display a user-friendly value. But there, we wouldn't need to care about the int64 conversion since we'd still have the BSON type to report alongside it.

Reply:
I agree. At the moment, `get_debug_info` also displays a `Document` or `PackedArray` instance, which isn't helpful to the user. In those cases, I'd rather use `phongo_bson_value_to_zval_legacy` to convert documents and arrays to their PHP representations. It will mean that I have to either add a flag to `phongo_bson_value_to_zval` (to indicate whether documents and arrays should be returned as BSON types) or find a better name than `_legacy`, as the latter indicates that the method will be dropped eventually.

I'll implement serialisation and then see what can be refactored. 👍

Reply:
Serialisation rules have been updated. For `Document`, `PackedArray`, and `Value`, the debug representation now includes raw data as well as the PHP representation. The type that is returned is a formatted string for convenience. Take the following example:
```
var_dump(Value::fromPHP([0, 1, 2]);
// Output
object(MongoDB\BSON\Value)#1 (2) {
  ["type"]=>
  string(5) "array"
  ["value"]=>
  object(MongoDB\BSON\PackedArray)#1 (2) {
    ["data"]=>
    string(36) "GgAAABAwAAEAAAAQMQACAAAAEDIAAwAAAAA="
    ["value"]=>
    array(3) {
      [0]=>
      int(1)
      [1]=>
      int(2)
      [2]=>
      int(3)
    }
  }
}
```

When serialising a `Document` or `PackedArray` instance, only the raw data is serialised. When serialising a `Value` instance, the result includes the type as well as the correct BSON instance. Take an Int64 for example:
```
var_dump(serialize(Value::fromPHP(new Int64(2))));
// Output
string(110) "O:18:"MongoDB\BSON\Value":2:{s:4:"type";i:18;s:5:"value";O:18:"MongoDB\BSON\Int64":1:{s:7:"integer";s:1:"2";}}"
```

The same is true for var_export:
```
var_export(Value::fromPHP(new Int64(2)), true);
// Output
\MongoDB\BSON\Value::__set_state(array(
   'type' => 18,
   'value' => \MongoDB\BSON\Int64::__set_state(array(
     'integer' => '2',
  )),
))
```